File: plugins/easyar_flutter/android/src/main/kotlin/com/netbil/easyar_flutter/EasyArFlutterPlugin.kt
package com.netbil.easyar_flutter

import io.flutter.embedding.engine.plugins.FlutterPlugin

/**
 * Plugin raiz do easyar_flutter.
 *
 * Registra a [EasyArViewFactory] para criar PlatformViews do tipo
 * `com.netbil/easyar_view` quando o Dart pedir (`AndroidView(viewType: ...)`).
 *
 * Fase 1: a factory retorna um placeholder View com texto "Inicializando
 * EasyAR...". Nada da engine nativa e chamado ainda — o objetivo e so
 * validar que a ponte Flutter <-> nativo funciona e que o `EasyAR.aar`
 * esta corretamente linkado no build.
 */
class EasyArFlutterPlugin : FlutterPlugin {
    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        binding.platformViewRegistry.registerViewFactory(
            "com.netbil/easyar_view",
            EasyArViewFactory(binding.binaryMessenger)
        )
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        // Na Fase 2 aqui vai o Engine.dispose() ou equivalente.
    }
}

File: plugins/easyar_flutter/ios/Classes/EasyArFlutterPlugin.swift
import Flutter
import UIKit

/// Plugin raiz do easyar_flutter no iOS.
///
/// Fase 1: so registra a factory de PlatformView. A factory retorna
/// uma UIView com label "EasyAR — Fase 1". Nenhuma chamada real ao
/// SDK EasyAR acontece ainda — objetivo e validar que o xcframework
/// esta linkado no build e a ponte Flutter <-> Swift funciona.
///
/// Fase 2: inicializar `easyar.Engine` com a chave Sense, abrir camera,
/// renderizar preview dentro da UIView via Metal/EAGL.
public class EasyArFlutterPlugin: NSObject, FlutterPlugin {

    public static func register(with registrar: FlutterPluginRegistrar) {
        let factory = EasyArViewFactory(messenger: registrar.messenger())
        registrar.register(factory, withId: "com.netbil/easyar_view")
    }
}
